src/test/java/ua/goit/timonov/enterprise/module_6_2/dao/jdbc/JdbcStorageDAOTest.java

```diff
@@ -0,0 +1,211 @@
+public class JdbcStorageDAOTest {
+
+    private DbController dbController;
+    private StorageDAO storageDAO;
+
+    @Autowired(required = true)
+    public void setDbController(DbController dbController) {
+        this.dbController = dbController;
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        dbController.deleteAllData();
+        dbController.fillTableJobs();
+        storageDAO = dbController.getStorageDAO();
+    }
+
+    @Test
+    public void testGetAllNormal() throws Exception {
+        List<Ingredient> createdIngredients = fillTableIngredient();
+        List<Ingredient> gotFromDbIngredients = storageDAO.getAll();
+        assertEqualsIngredientLists(createdIngredients, gotFromDbIngredients);
+    }
+
+    @Test
+    public void testGetAllAbnormal_1() throws Exception {
+        List<Ingredient> createdIngredients = fillTableIngredient();
+        Ingredient redPepper = new Ingredient("red pepper", 2000);
+        storageDAO.add(redPepper);
+        List<Ingredient> gotFromDbIngredientes = storageDAO.getAll();
+        assertNotEqualsIngredientLists(createdIngredients, gotFromDbIngredientes);
+    }
+
+    @Test
+    public void testGetAllAbnormal_2() throws Exception {
+        List<Ingredient> createdIngredients = new ArrayList<>();
+        List<Ingredient> gotFromDbIngredients = storageDAO.getAll();
+        assertEquals(createdIngredients, gotFromDbIngredients);
+    }
+
+    @Test
+    public void testAddNormal() throws Exception {
+        List<Ingredient> createdIngredients = fillTableIngredient();
+
+        Ingredient redPepper = new Ingredient("red pepper", 3000);
+        createdIngredients.add(redPepper);
+        storageDAO.add(redPepper);
+        List<Ingredient> gotFromDbIngredients = storageDAO.getAll();
+        assertEqualsIngredientLists(createdIngredients, gotFromDbIngredients);
+
+        Ingredient eggplant = new Ingredient("eggplant", 15000);
+        createdIngredients.add(eggplant);
+        storageDAO.add(eggplant);
+        gotFromDbIngredients = storageDAO.getAll();
+        assertEqualsIngredientLists(createdIngredients, gotFromDbIngredients);
+    }
+
+    @Test
+    public void testAddAbnormal() throws Exception {
+        List<Ingredient> createdIngredients = fillTableIngredient();
+
+        Ingredient redPepper = new Ingredient("red pepper", 2000);
+        createdIngredients.add(redPepper);
+        storageDAO.add(redPepper);
+        List<Ingredient> gotFromDbIngredients = storageDAO.getAll();
+        assertEqualsIngredientLists(createdIngredients, gotFromDbIngredients);
+
+        Ingredient eggplant = new Ingredient("eggplant", 15000);
+        createdIngredients.add(redPepper);
+        storageDAO.add(eggplant);
+        gotFromDbIngredients = storageDAO.getAll();
+        assertNotEqualsIngredientLists(createdIngredients, gotFromDbIngredients);
+    }
+
+    @Test
+    public void testSearchByIdNormal() throws Exception {
+        restoreAllData();
```

Comment:
what does restoreAllData();  do? I think it`s better to move it to @Before section